src/ImageSharp/Formats/Jpeg/Components/Encoder/RgbToYCbCrConverterVectorized.cs

```diff
@@ -0,0 +1,182 @@
+            var fn0331264 = Vector256.Create(-0.331264f);
+            var f128 = Vector256.Create(128f);
+            var fn0418688 = Vector256.Create(-0.418688f);
+            var fn0081312F = Vector256.Create(-0.081312F);
+            var f05 = Vector256.Create(0.5f);
+
+            ref Vector256<byte> inRef = ref Unsafe.As<Rgb24, Vector256<byte>>(ref MemoryMarshal.GetReference(rgbSpan));
+
+            for (int i = 0; i < 2; i++)
+            {
+                ref Vector256<float> destYRef = ref Unsafe.Add(ref Unsafe.As<Block8x8F, Vector256<float>>(ref yBlock), i * 4);
+                ref Vector256<float> destCbRef = ref Unsafe.Add(ref Unsafe.As<Block8x8F, Vector256<float>>(ref cbBlock), i * 4);
+                ref Vector256<float> destCrRef = ref Unsafe.Add(ref Unsafe.As<Block8x8F, Vector256<float>>(ref crBlock), i * 4);
+
+                Vector256<byte> red = Unsafe.Add(ref inRef, i * 3);
+                Vector256<byte> green = Unsafe.Add(ref inRef, (i * 3) + 1);
+                Vector256<byte> blue = Unsafe.Add(ref inRef, (i * 3) + 2);
+
+                for (int j = 0; j < 2; j++)
+                {
+                    // 1st part of unrolled loop
+                    Vector256<byte> mask = Unsafe.Add(ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(ExtractionMasks)), j * 2);
+
+                    Vector256<float> r = Avx.ConvertToVector256Single(Avx2.Shuffle(red, mask).AsInt32());
+                    Vector256<float> g = Avx.ConvertToVector256Single(Avx2.Shuffle(green, mask).AsInt32());
+                    Vector256<float> b = Avx.ConvertToVector256Single(Avx2.Shuffle(blue, mask).AsInt32());
+
+                    // (0.299F * r) + (0.587F * g) + (0.114F * b);
+                    Vector256<float> yy0 = Fma.MultiplyAdd(f0299, r, Fma.MultiplyAdd(f0587, g, Avx.Multiply(f0114, b)));
+
+                    // 128F + ((-0.168736F * r) - (0.331264F * g) + (0.5F * b))
+                    Vector256<float> cb0 = Avx.Add(f128, Fma.MultiplyAdd(fn0168736, r, Fma.MultiplyAdd(fn0331264, g, Avx.Multiply(f05, b))));
+
+                    // 128F + ((0.5F * r) - (0.418688F * g) - (0.081312F * b))
+                    Vector256<float> cr0 = Avx.Add(f128, Fma.MultiplyAdd(f05, r, Fma.MultiplyAdd(fn0418688, g, Avx.Multiply(fn0081312F, b))));
+
+                    // 2nd part of unrolled loop
+                    mask = Unsafe.Add(ref Unsafe.As<byte, Vector256<byte>>(ref MemoryMarshal.GetReference(ExtractionMasks)), (j * 2) + 1);
+
+                    r = Avx.ConvertToVector256Single(Avx2.Shuffle(red, mask).AsInt32());
+                    g = Avx.ConvertToVector256Single(Avx2.Shuffle(green, mask).AsInt32());
+                    b = Avx.ConvertToVector256Single(Avx2.Shuffle(blue, mask).AsInt32());
+
+                    // (0.299F * r) + (0.587F * g) + (0.114F * b);
+                    Vector256<float> yy1 = Fma.MultiplyAdd(f0299, r, Fma.MultiplyAdd(f0587, g, Avx.Multiply(f0114, b)));
```

Comment:
We have a `SimdUtils.HWIntrinsics.MultiplyAdd` method that can be used to abstract this away. It falls back to `Avx.Add/Multiply`

Reply:
All processors that support AVX2 also support FMA, so it's fine to require both explicitly.  As long as you protect against use with the JIT flags disabling FMA by including the IsSupported check, it won't make any difference to real users.  Just comes down to a matter of preference.

Reply:
Switched to `SimdUtils.HWIntrinsics.MultiplyAdd`